src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,39 @@
+// child class of Popup
+
+import Popup from "./Popup.js";
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleFormSubmit, confirmationButtonSelector) {
+    console.log(popupSelector);
+    super(popupSelector);
+    this._handleFormSubmit = handleFormSubmit;
+    this._confirmationButton = this.popup.querySelector(
+      confirmationButtonSelector
+    );
+  }
+
+  setId(id) {
+    this._id = id;
+  }
+
+  getId() {
+    return this._id;
+  }
+
+  setCard(card) {
```

Comment:
`PopupWithConfirmation` is a general class. It can be used for any confirmation, not only card delete confirmation. This class should have a setter for `_handleFormSubmit`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

this._form.addEventListener("submit", (evt) => { 
  evt.preventDefault(); 
  this._handleFormSubmit(); 
})
```